pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -327,6 +337,21 @@ func shouldRefresh(es esv1alpha1.ExternalSecret) bool {
 	return !es.Status.RefreshTime.Add(es.Spec.RefreshInterval.Duration).After(time.Now())
 }
 
+// isSecretValid checks if the secret exists, and it's data is consistent with the calculated hash.
+// todo: test coverage
+func isSecretValid(existingSecret v1.Secret) bool {
+	// if target secret doesn't exist, we need to refresh
+	if existingSecret.UID == "" {
+		return false
+	}
+
+	// if the calculated hash is empty, or is different from the calculation, then it's invalid
+	if existingSecret.Annotations == nil || existingSecret.Annotations[annotationDataHash] != utils.ObjectHash(&existingSecret) {
```

Comment:
I think it should compare with `.Data` here:

```suggestion
	if existingSecret.Annotations == nil || existingSecret.Annotations[annotationDataHash] != utils.ObjectHash(existingSecret.Data) {
```